xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -813,25 +893,22 @@ void CGUIDialogPVRTimerSettings::DaysFiller(
   {
     list.clear();
     current = 0;
+    CDateTime time = pThis->m_minDateCovered;
 
-    // Data range: "today" until "yesterday next year"
-    const CDateTime now(CDateTime::GetCurrentDateTime());
-    CDateTime time(now.GetYear(), now.GetMonth(), now.GetDay(), 0, 0, 0);
-    const CDateTime yesterdayPlusOneYear(
-      time.GetYear() + 1, time.GetMonth(), time.GetDay() - 1, time.GetHour(), time.GetMinute(), time.GetSecond());
-
-    while (time <= yesterdayPlusOneYear)
+    list.push_back(std::make_pair("< " + time.GetAsLocalizedDate(), pThis->GetDateAsIndex(time - CDateTimeSpan(1, 0, 0, 0))));
```

Comment:
Wouldn't it be a good idea to add the > and < entries only if start time and|or end time are out of range? What happens if start and|or end time of the timer are not out of range and the user selects one of the two "special" entries? Will the user understand, that "< day x" then actually means "day x - 1"? In the case of one or both values are actually out of range (but heaving a concrete value that is not displayed here), will the user understand that "< day x" actually means some other concrete day (that is not displayed here)? 

The longer I think about the > and < entry idea, the less I tend to like it.

My proposal: Leave this stuff out. It makes the code complicated and users will not really understand the "feature" (at least from my point of view). If a value is out of range, add just one extra entry for exactly this value here, labeled with the concrete out of range day.